test/Basic/matrix_single_subscript_load.test

```diff
@@ -0,0 +1,71 @@
+#--- source.hlsl
+RWBuffer<int> In : register(u0);
+RWBuffer<int> Out : register(u1);
+
+[numthreads(1,1,1)]
+void main() {
+    int4x4 A = int4x4(In[0],  In[1],  In[2],
+                      In[3],  In[4],  In[5],
+                      In[6],  In[7],  In[8],
+                      In[9],  In[10], In[11],
+                      In[12], In[13], In[14],
+                      In[15]);
+    
+    for (int i = 0; i < 4; i++) {
+        int4 B;
+        if (i % 2 == 0)
+            B = A[i];
+        else if (i % 3 == 0)
+            B = A[i].abgr;
```

Comment:
Note: `abgr` is also a symmetric swizzle. `bagr` is another asymmetric swizzle you could use.